assets/app/view/game/buy_trains.rb

```diff
@@ -227,9 +227,12 @@ def remaining_trains
           },
         }
 
-        rows = @depot.upcoming.group_by(&:name).flat_map do |name, trains|
-          [h(:div, name),
-           h(:div, @game.format_currency(trains.first.price)),
+        rows = @depot.upcoming.group_by(&:name).flat_map do |_, trains|
+          train = trains.first
+          names =  train.variants.values.map { |var| var[:name] }
```

Comment:
can you create a helper function in train

def names_and_prices
[names, prices]
end

and refactor these two places where you do this pattern?